### PR TITLE
Add documentation about updating H2O LLM Studio

### LINKED ISSUE #150: [FEATURE] Explain where the data is stored
### 🚀 Feature

We should explain where data is stored to prevent unexpected data loss
- sql database
- datasets
- trained models
- cache folder

We should also explain how these can be modified to personal needs. 

One possible place for this would be the FAQ in the README.md

### Motivation

Prevent unexpected dataloss when upgrading.

### PR DESCRIPTION
**Pull request by maxjeblick**
This PR adds the possibility to change the output and data folder in the config, as well as updating the Readme w.r.t. changing those folders and how to update the app.

Fixes #150

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -331,6 +331,35 @@ There are various parameters that can be tuned while keeping a specific LLM back
...
+
+It is possible to change the default folders `data` and `output` in [app_utils/config.py](app_utils/config.py) 
+(change `data_folder` and `output_folder`).
+
+> ❓ How can I update H2O LLM Studio?
+
+To update H2O LLM Studio, you have two options:
+
+1. Using the latest main branch: Execute the commands `git checkout main` and `git pull` to obtain the latest updates from the main branch.
+2. Using the latest release branch: Execute the commands `git pull` and `git checkout v0.0.3` (replace 'v0.0.3' with the desired version number) to switch to the latest release branch.
```
Comment (pascal-pfeiffer) [inline README.md:350]: release "tag", we don't have actual branches for them.

**Review by pascal-pfeiffer — CHANGES_REQUESTED** [review]
Thank you! I think this should cover for now. 
two comments in the code. I think, that we should ensure full downward compatibility only starting from 1.0.0

File: README.md [hunk 2]
```diff
@@ -331,6 +331,35 @@ There are various parameters that can be tuned while keeping a specific LLM back
...
+
+H2O LLM Studio follows the practices of [Semantic Versioning 2.0.0](https://semver.org/),
+ensuring backward compatibility for all minor versions.
+That is, after updating to a newer minor version of H2O LLM Studio, you will be able to access and view your previous data and experiments.
+The update process does not remove or erase any existing data folders or experiment records.
+This means that all your old data, including the user database, uploaded datasets, and experiment results, 
+will still be available to you within the updated version of H2O LLM Studio.
```
[the remark this answers is not in the document]
Comment (maxjeblick) [reply README.md]: I removed the corresponding section.

**Review by pascal-pfeiffer — APPROVED** [review]
lgtm

**Pull request merged**